examples/flutter_firebase_login/packages/authentication_repository/lib/src/authentication_repository.dart

```diff
@@ -89,6 +96,55 @@ class AuthenticationRepository {
     }
   }
 
+  /// Starts the Sign In with Apple Flow.
+  ///
+  /// Throws a [logInWithApple] if an exception occurs.
+  Future<void> logInWithApple() async {
+    // To prevent replay attacks with the credential returned from Apple, we
+    // include a nonce in the credential request. When signing in with
+    // Firebase, the nonce in the id token returned by Apple, is expected to
+    // match the sha256 hash of `rawNonce`.
+    final rawNonce = generateNonce();
+    final nonce = sha256ofString(rawNonce);
+
+    try {
+      final appleCredential = await SignInWithApple.getAppleIDCredential(
+        scopes: [
+          AppleIDAuthorizationScopes.email,
+          AppleIDAuthorizationScopes.fullName,
+        ],
+        nonce: nonce,
+      );
+
+      // Create an `OAuthCredential` from the credential returned by Apple.
+      final credential = firebase_auth.OAuthProvider('apple.com').credential(
+        idToken: appleCredential.identityToken,
+        rawNonce: rawNonce,
+      );
+
+      await _firebaseAuth.signInWithCredential(credential);
+    } on Exception {
+      throw LogInWithAppleFailure();
+    }
+  }
+
+  /// Generates a cryptographically secure random nonce, to be included in a
+  /// credential request.
+  String generateNonce([int length = 32]) {
+    final charset =
+        '0123456789ABCDEFGHIJKLMNOPQRSTUVXYZabcdefghijklmnopqrstuvwxyz-._';
+    const random = Random.secure();
+    return List.generate(length, (_) => charset[random.nextInt(charset.length)])
+        .join();
+  }
+
+  /// Returns the sha256 hash of [input] in hex notation.
+  String sha256ofString(String input) {
```

Comment:
Thinking about having all the crypto functionality on a separate package and class, to make it reusable and make a Crypto API a dependency of AuthenticationRepository. What do you think?

Reply:
Sounds good. I will fix both suggestions in the MR.

Reply:
done